[1 - 2단계 방탈출 예약 대기] 랜디(김정우) 미션 제출합니다.

KJungW opened the pull request:
<!-- 

## 코드 리뷰 팁

- 코드와 관련된 질문이 있다면, PR 본문에 적기 보다는 해당 코드를 선택하고 코멘트를 남겨주세요.
  - [참고: Adding comments to a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-comments-to-a-pull-request)

-->
## 체크 리스트
- [x] 미션의 필수 요구사항을 모두 구현했나요?
- [x] Gradle `test`를 실행했을 때, 모든 테스트가 정상적으로 통과했나요?
- [x] 애플리케이션이 정상적으로 실행되나요?
- [x] [프롤로그](https://prolog.techcourse.co.kr)에 셀프 체크를 작성했나요?
  <!-- 작성한 셀프 체크의 링크를 남겨주세요. -->
  - https://prolog.techcourse.co.kr/studylogs/4361

## 객체지향 생활체조 요구사항을 얼마나 잘 충족했다고 생각하시나요?
### 1~5점 중에서 선택해주세요.

- [ ] 1 (전혀 충족하지 못함)
- [x] 2
- [ ] 3 (보통)
- [ ] 4
- [ ] 5 (완벽하게 충족)

### 선택한 점수의 이유를 적어주세요.
<!-- 이유 작성 -->
아직 부족한 부분이 많습니다! 리펙토링을 많이 해보겠습니다!

## 베이스 코드 선택 체크
- [ ] 이전 미션의 내 코드에서 시작 
- [x] 이전 미션의 페어의 코드에서 시작

## 어떤 부분에 집중하여 리뷰해야 할까요?
<!-- 리뷰어가 효과적으로 피드백할 수 있도록 중점적으로 리뷰 받고 싶은 내용을 *요약* 형태로 작성해 주세요.
리뷰해야 할 포인트를 요약해 강조해 주시면, 리뷰어가 코드 전체를 이 부분에 집중해 리뷰할 수 있습니다.
반면 특정 코드부분에 대한 피드백이 필요하다면, 이 곳에 적기 보다 해당 코드를 선택하고 코멘트를 남기는 것을 권장합니다. -->

파랑, 안녕하세요! 저번에 장기 미션을 진행할 때 파랑께 정말 좋은 피드백을 많이 받아서 도움이 많이 되었습니다. 감사합니다! 이번 리뷰도 잘부탁드립니다!

이번 미션은 아직 Merge되지 않은 이전 미션 코드로 시작하게 되면서 아직 정리할 점이나 고쳐야 할 점이 많을 것 같습니다! 이번 미션에서 추가된 점을 중점적으로 리뷰해주시면 해당 부분을 수정하면서 기존의 코드 빠르게 리펙토링해보겠습니다! 항상 좋은 리뷰 감사합니다!

[현재 미션에서 생성된 커밋 범위](https://github.com/woowacourse/spring-roomescape-waiting/pull/201/files/76043ec167b54c6d04314c863a6282eef6e45997..5685a22b8a671c2ea741eeaf3a31e310c00c7fe8)

### 📌 코드리뷰 질문 그라운드룰 (지우지 마세요)
✅ 많은 사람과 공유하고 토론해보는 것이 더 좋은 질문에는, 리뷰어가 💬말풍선 이모지를 남겨 토론을 추천할 수 있어요.  
✅ 토론을 추천받은 질문은 `#be-7기-잡담-질답`슬랙채널이나 캠퍼스에서 자유롭게 토론을 이어가보세요!  
✅ 코드에 대한 맥락 없이 리뷰어의 주관적인 의견을 묻거나, 단순히 현업에서는 어떻게 하는지를 묻는 질문은 코드 리뷰 본문에는 어울리지 않아요.  
코드 리뷰는 변경된 코드의 목적, 의도, 품질을 중심으로 논의하는 자리예요.  
✅ 코드와 관련된 질문이 있다면 PR 본문에 적기보다는, 해당 코드를 선택해 코멘트를 남겨주세요.

2Jin1031 committed:
init: 이전 미션 코드 사용

Commit:
refactor: JPA 의존성 추가 및 엔티티 변형

2Jin1031 committed:
refactor: 예약 시간에서 Jdbc 방식을 Jpa로 수정

Commit:
refactor: 테마에서 Jdbc 방식을 Jpa로 수정

Commit:
refactor: 회원에서 Jdbc 방식을 Jpa로 수정

2Jin1031 committed:
refactor: 유저에서 Jdbc 방식을 Jpa로 수정

2Jin1031 committed:
docs: README 기능 목록 업데이트

Commit:
refactor: 빈 생성자 생성

Commit:
test: 테스트 데이터 세팅 방식 변경

Commit:
feat: 자신의 예약 리스트 조회 기능 구현

2Jin1031 committed:
feat: ID 전략 수정 및 추가 필드 sql 적용

Commit:
fix: 자신의 예약 조회 페이지 코드 수정

Commit:
test: 테스트 컴파일 오류 수정

Commit:
test: 회원별 예약 조회 레파지토리 테스트

2Jin1031 committed:
refactor: 서비스 계층에 @Transactional 추가

2Jin1031 committed:
test: 유저 서비스 계층 테스틑 추가

2Jin1031 committed:
refactor: 사용하지 않는 클래스 제거

2Jin1031 committed:
test: 멤버 controller 테스트 추가

2Jin1031 committed:
docs: README 업데이트

summerlunaa commented on src/main/java/roomescape/member/controller/MemberController.java:
```diff
@@ -24,4 +26,10 @@ public ResponseEntity<List<UserResponseDto>> findAll() {
+    public ResponseEntity<List<ReservationWithBookStateDto>> findAllReservationsByMember(User member) {
+        List<ReservationWithBookStateDto> reservations = memberService.findAllReservationByMember(member);
+        return ResponseEntity.status(HttpStatus.OK).body(reservations);
+    }
```
회원의 **예약**을 가져오는 기능이 MemberController에 있네요! 회원의 예약을 가져오는 기능, 리뷰를 가져오는 기능, 결제내역을 가져오는 기능이 생기면 모두 MemberController에 들어가야 할까요? Member 도메인이 다른 모든 도메인을 의존하게 되는데 문제는 없을까요? 좀 더 고민해보시죠~

KJungW replied:
확실히 '회원의 예약을 가져오는 기능'의 주체는 리턴값인 예약이고 어떤 회원인지는 그저 필터값이라는 생각이 듭니다! MemberService로 해당 기능을 옮겨보겠습니다!

summerlunaa replied:
엇 여전히 해당 API가 `MemberController`에 있는데 서비스만 옮기신 걸까요 ㅎㅎㅎ

KJungW replied:
앗.. 회원의 예약을 조회하는 기능이 회원서비스에 있었던 점이 이전부터 마음에 걸려서 서비스 레벨로만 생각했습니다! 컨트롤러에서도 마찬가지로 예약을 조회한다는 점이 중점이기 때문에 회원 컨트롤러가 아닌 예약 컨트롤러로 기능을 옯겨보겠습니다!

summerlunaa commented on src/main/java/roomescape/member/domain/dto/ReservationWithBookStateDto.java:
```diff
@@ -0,0 +1,24 @@
+public record ReservationWithBookStateDto(
+        Long id,
+        LocalDate date,
+        String statusText,
```
statusText 필드명이 애매하네요. 어떤 상태인가요? Text는 왜 붙었나요?

KJungW replied:
Enum을 문자열로 보낸다는 의미에서 관성적으로 붙인 것 같습니다! 말씀해주신대로 애매하다는 생각이 들어 바로 수정하겠습니다!

summerlunaa commented on src/main/java/roomescape/reservation/domain/ReservationStatus.java:
```diff
@@ -0,0 +1,15 @@
+package roomescape.reservation.domain;
+
+public enum ReservationStatus {
+    BOOKED("예약");
```
Enum 사용 좋습니다 👍

KJungW replied:
감사합니다!

summerlunaa commented on src/main/java/roomescape/admin/service/AdminService.java:
```diff
@@ -12,6 +13,7 @@
 import roomescape.user.service.UserService;
 
 @Service
+@Transactional
```
`@Transactional` 애노테이션은 왜 붙이셨나요? 어떤 역할을 하나요?

KJungW replied:
@Transaction은 하나의 서비스 로직연산을 하나의 트랜젝션으로 관리하게 해주는 애노테이션입니다. 서비스에 @Transaction을 적용함으로써 서비스의 각 로직들을 하나의 트랜젝션으로 관리할 수 있고, 이를 통해 중간에 예상치 못한 예외 등이 발생해서 일부 쿼리의 결과만 DB에 적용되는 불상사를 막을 수 있습니다.

현재 단계에서는 필요하지 않다고 생각해 세부 옵션에 대해 공부하지는 않았지만 이번 리뷰를 기다리면서 readOnly를 활용환 최적화, 트랜젝션 전파, 격리 수준에 대해 추가 학습을 진행해볼 예정입니다!

summerlunaa commented on src/main/java/roomescape/reservation/domain/Reservation.java:
```diff
@@ -1,32 +1,55 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private LocalDate date;
```
<img width="333" alt="image" src="https://github.com/user-attachments/assets/ef328069-908d-47cb-93e4-873a6e256632" />

`@Column`이라는 애노테이션도 있어요. 해당 애노테이션이 어떤 역할을 하고, 어떤 설정을 추가할 수 있는지 공부해보면 좋을 것 같습니다 :)

KJungW replied:
좋은 학습 키워드 감사합니다. 학습을 진행하고 엔티티에 적용해보겠습니다!

@Column은 엔티티의 필드와 테이블의 컬럼을 명시적으로 매핑하는 애노테이션이라고 학습을 진행했습니다!
- name : 필드와 매핑할 컬럼 이름 지정
- insertable : 삽입 연산에 해당 컬럼을 포함할지 여부 지정
- updateable : 업데이트 연산에 해당 컬럼을 포함할지 여부 지정
- nullable : 해당 컬럼에 대한 not null 제약 조건 추가
- unique : 해당 컬럼에 대한 유니크 제약 조건 추가
- length : 문자 길에에 대한 제약 조건 추가

summerlunaa commented on src/main/java/roomescape/reservation/domain/Reservation.java:
```diff
@@ -1,32 +1,55 @@
 
-    public Reservation(Long id, LocalDate date, ReservationTime reservationTime, Theme theme, User user) {
+    @ManyToOne(fetch = FetchType.LAZY)
+    private ReservationTime reservationTime;
```
기본값인 FetchType.EAGER 대신 LAZY를 사용해주셨네요! 두 설정의 차이점은 무엇인가요? 왜 LAZY를 사용하셨나요?

KJungW replied:
FetchType.EAGER는 즉시로딩으로 Reservation을 DB에서 가져올때 연관 관계의 엔티티를 자동으로 조인해서 함께 가져오는 속성이고, FetchType.LAZY는 지연로딩으로 Reservation을 DB에서 가져올 때 연관 관계의 엔티티를 조인하지 않지 않고 가져온 뒤에 나중에 필요하면 로딩하는 것을 의미합니다!

FetchType.EAGER를 사용할 경우 조회할 엔티티의 연관 관계 엔티티를 즉시 로딩하고 또 연관 관계 엔티티의 연관 관계 엔티티를 즉시 로딩하는 등의 위험한 연쇄작용이 발생할 수 있어 디폴트로 FetchType.Lazy 전략을 사용했습니다. 

사실 FetchType.Lazy도  엔티티의 모든 필드를 사용할 경우 쿼리가 여러번 발생해 비효율적이라는 단점도 있지만 이 부분은 나중에 성능적으로 문제가 생기면 fetch join이나 객체 그래프를 활용해 최적화 해보겠습니다!

summerlunaa commented on src/main/java/roomescape/reservation/service/ReservationService.java:
```diff
@@ -44,17 +47,22 @@ public List<ReservationResponseDto> findAll() {
         repository.deleteById(id);
     }
 
+    private Reservation findByIdOrThrow(Long id) {
```
참고) 보통 확정적으로 값을 가져오는 경우는 get, 있을수도 없을수도 있는 경우는 find를 사용하더라구요

KJungW replied:
앗.. 확실히 컨벤션이 잘 지켜지지 않은 것 같습니다! 바로 수정하겠습니다!

summerlunaa replied:
load는 어떤 의미의 네이밍일까요? 데이터 조회 메서드 네이밍으로는 처음 보는 단어라 어색하게 느껴져서 여쭤봐요 😅

KJungW replied:
컨트롤러에서 호출하기 위해 만든 서비스의 public 메서드의 경우 get~~() 컨벤션을 사용했고, 서비스에서 내부적으로 로직 처리를 위해 DB로부터 값을 확정적으로 로딩하기 위해 만든 서비스의 private 메서드의 경우 load~~()를 사용했습니다! 

두 개의 역할이 다르다고 생각해 각각 별도의 컨벤션으로 관리하는 편이 가독성을 높일 수 있다고 생각했고, 이때 DB로부터 데이터를 로딩해온다는 의미로 load()를 사용했습니다. 

하지만 짚어주신 부분을 다시 생각해보니 확실히 load라는 컨벤션이 일반적이지 않아 다른 개발자가 그 의미를 파악하기 쉽지않고 이로인해 원래의 사용의도와 달리 가독성이 오히려 떨어진다는 생각이 들었습니다. 오히려 get 컨벤션을 사용해서 메서드의 의미를 정확하게 전달하는 편이 가독성을 높일 수 있을 것 같다는 생각이 듭니다! 바로 수정하겠습니다!

summerlunaa commented on src/main/java/roomescape/theme/domain/Theme.java:
```diff
@@ -33,4 +44,18 @@ public String getDescription() {
+            return false;
+        }
+        Theme theme = (Theme) o;
+        return Objects.equals(id, theme.id);
```
id만 같으면 같은 객체일까요? id가 null이면 어떻게 되나요? equals&hashCode가 재정의된 곳과 아닌 곳(Reservation)의 차이는 무엇인가요?

KJungW replied:
엔티티는 도메인 개념을 구현하는 실제 개체로써 id를 통해 식별된다는 주요 특징을 가지고 있습니다. 그렇기에 엔티티의 경우 id가 동일하면 동일한 객체라고 생각하고 id가 null인 경우에는 아직 엔티티 객체로 취급할 수 없다고 생각합니다!

그렇기에 id가 null인 경우에는 비교가 불가능하다는 생각이 들었고, 직접 테스트 코드를 통해 id가 null인 경우의 equals()가 어떻게 작동하는지 파악했습니다. 결과는 둘다 null인 경우 true가 나온다는 것을 알게되었고, 이를 비교 불가능하다는 의미에서 false를 리턴하는 방향으로 수정했습니다! 생각하지 못한 측면인데 피드백 감사합니다!
(equals&hashCode가 재정의되지 않은 곳은 빠뜨린 부분입니다! 바로 수정하겠습니다!)

summerlunaa commented on src/main/java/roomescape/user/domain/User.java:
```diff
@@ -1,19 +1,42 @@
+import java.util.Set;
+import roomescape.reservation.domain.Reservation;
+
+@Entity(name = "users")
```
왜 user만 복수형인가요?

summerlunaa replied:
member 와 user는 뭐가 다른가요?..

KJungW replied:
**엔티티명을 users로 한 이유**
- 엔티티명을 user로 하면 user테이블에 생성되는 DB에 user라는 기본 테이블에 이미 존재해서 에러가 발생했습니다. 그래서 엔티티명을 약간 수정해서 user가 아닌 users 테이블이 생성되도록 함으로써 해당 문제를 해결했습니다.
- 하지만 지금 다시 생각해보니 User만 복수형이라는 점이 확실히 통일감이 없다는 생각이 듭니다. 엔티티명을 Member로 수정하겠습니다!

**member와 user의 다른점**
- 저도 확실히 헷갈릴만한 모호한 부분이라고 생각합니다! 그래서 회원 엔티티명을 Member로 하고, 권한명을 (GENERAL,  ADMIN)으로 변경하겠습니다!

summerlunaa replied:
<img width="291" alt="image" src="https://github.com/user-attachments/assets/9cc597d5-f323-4c22-8794-db3beb08cffa" />

좋습니다! 다만 Member 라는 단어를 사용하기로 했다면 User라는 단어는 프로젝트 내에서 완전히 제거하는 게 좋을 것 같네요ㅎㅎㅎ

KJungW replied:
앗.. 거기까지 생각하지 못했습니다! 놓친 부분을 집어주셔서 감사합니다. 바로 수정하겠습니다!

summerlunaa commented on src/main/java/roomescape/user/domain/User.java:
```diff
@@ -1,19 +1,42 @@
+    private String password;
+
+    @OneToMany(mappedBy = "user")
+    private Set<Reservation> reservations;
```
`@OneToMany`를 사용하셨군요! 유저가 reservations를 갖도록 설계한 이유가 궁금합니다!

KJungW replied:
이번 미션에서 구현하는 기능이 특정 회원의 예약리스트를 조회하는 기능이었고, 해당 기능은 자주 사용될 수 있는 기능이라고 생각했습니다. 그래서 Member와 Reservation 사이에 일대가 양방향 관계를 만들고 조회를 간편하게 수행함으로써 유지보수나 가독성 측면에서 이점이 있다고 생각했습니다.

하지만 리펙토링 과정에서 이점에 대해 다시 곰곰히 생각을 해보았고 결과적으로는 다음과 같은 이유로 양방향 관계를 제거하게 되었습니다!
-  양방향 관계에서 연관관계의 주인이 아닌 측에서는 수정이 불가능하다는 점을 개발자가 알고 있어야 한다는 점이 아쉬웠습니다!
- 양방향 관계를 활용한 로딩이 아닌 레파지토리를 통해 명시적으로 조회하는 것이 오히려 내부 로직이나 쿼리 로그를 쉽게 확인할 수 있기에 가독성이 올라간다고 판단했습니다!

summerlunaa replied:
넵넵 OneToMany는 개발자가 다루기 까다롭기 때문에 사용 했을 때 명확한 이점이 있지 않은 이상 웬만하면 지양하는 것 같습니다. (근데 OneToMany를 직접 사용해보고 단점을 몸소 체험해보는 것도 도움이 되더라구요ㅋㅋ)

KJungW replied:
의도치않게 주변으로부터 다음미션 내용을 듣게 되었습니다. 다음 미션에서 대기 목록을 만들어 볼때  @OneToMany를 사용해보면서 단점을 직접 한번 체험해보겠습니다!

summerlunaa commented on src/main/resources/application.properties:
```diff
@@ -3,3 +3,6 @@ spring.datasource.url=jdbc:h2:mem:database
 security.jwt.token.secret-key=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIiLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1MTYyMzkwMjJ9.ih1aovtQShabQ7l0cINw4k1fagApg3qLWiB8Kt59Lno
 security.jwt.token.expire-length=3600000
+
+spring.jpa.hibernate.ddl-auto=create
+spring.jpa.defer-datasource-initialization=true
```
각각의 설정들은 어떤 의미/역할인가요?

KJungW replied:
"spring.jpa.hibernate.ddl-auto"의 경우 서버를 시작할 때 DB 스키마의 초기화 여부를 설정하는 옵션이라고 학습을 했습니다. 특히 create값의 경우 엔티티 정의에 맞게 스키마를 초기화 하겠다는 옵션입니다. 개발 단계에서는 편의를 위해 사용하지만 이후 서비스 단계에서는 사용하면 안되는 옵션입니다. 

"spring.jpa.defer-datasource-initialization"의 경우 DB 스키마를 초기화할때 초기 데이터를 삽입할지 여부를 설정하는 옵션이라고 알고있었는데 이번 기회에 제대로 학습하게 되었습니다. 해당 옵션은 data.sql와 같은 활용한 초기 데이터 삽입의 실행을 JPA의 실행 이후 시점으로 지연하는 옵션입니다. 실제로 해당 옵션을 false로 설정하면 JPA가 테이블을 생성하기도 전에 데이터 삽입 쿼리가 실행되어 에러가 발생하게 됩니다!

summerlunaa replied:
좋습니다! 다른 설정은 뭐가 있는지, 실제 운영되는 서비스라면 설정 값을 무엇으로 하는 게 좋을지 고민해보는 것도 좋을 것 같아요 👍

KJungW replied:
스프링 데이터 JPA 관련 주요 설정값들을 공부하고 정리해보았습니다. 그리고 정리한 것 중에서 다음의 옵션값이 현재 프로젝트에서도 유용할 것이라 생각해 적용해보았습니다!

- spring.jpa.show-sql=true
    - 실행되는 SQL문 출력
- spring.jpa.properties.hibernate.format_sql=true
    - 출력되는 SQL문을 알아보기 쉽게 정리해서 출력
- spring.jpa.open-in-view=false
    - 트랜젝션이 종료되면 영속성 컨텍스트도 닫도록 설정
    - 서비스에서 트랜젝션이 종료되었는데도 컨트롤러에서 지연로딩에 의한 LazyInitializationException이 발생하지 않아 그 이유가 궁금했었는데 해당 옵션이 원인이었다는 것을 알게되었습니다.
- spring.jpa.properties.hibernate.default_batch_fetch_size=50
    - 지연로딩이나 컬렉션을 가져올때 한번의 Select 쿼리로 몇개의 데이터를 일괄로 가져올지 정하는 옵션 (in절을 활용한 배치처리 옵션)
    - 지연로딩 시에 발생하는 N+1 문제를 방지하기 위해 적용했는데 아직 큰 이점이 느껴지지 않습니다! 다음 단계에서 @OneToMany를 사용해볼 예정인데 그때 한번 이점을 찾아보겠습니다!

summerlunaa commented on src/test/java/roomescape/member/controller/MemberControllerTest.java:
```diff
@@ -0,0 +1,92 @@
+    ) {
+        date = LocalDate.now().plusDays(1);
+
+        memberStatic = memberTestDataConfig.getSavedMember();
```
Repository를 사용하지 않고 테스트 설정용 Config 클래스를 저장된 데이터를 가져오는 역할로도 사용하는 이유가 있을까요?

KJungW replied:
페어와 여러 이야기를 나누면서 테스트 코드를 한눈에 파악하기 힘들다는 점에서 수정이 필요하다는 생간한 부분이었습니다. 페어프로그래밍 동안에는 미쳐 수정하지 못했지만 이번에 바로 리펙토링해보겠습니다!

summerlunaa commented on src/test/java/roomescape/reservationTime/repository/JdbcReservationTimeRepositoryTest.java:
```diff
@@ -6,19 +6,15 @@
-@JdbcTest
-@Import({JdbcReservationTimeRepository.class, KeyHolderManager.class})
+@DataJpaTest
 class JdbcReservationTimeRepositoryTest {
```
```suggestion
class ReservationTimeRepositoryTest {
```

KJungW replied:
앗.. 바로 수정하겠습니다!

summerlunaa commented on src/test/java/roomescape/reservation/repository/ReservationRepositoryTest.java:
```diff
@@ -0,0 +1,83 @@
+        MemberTestDataConfig.class,
+        ReservationTimeTestDataConfig.class
+})
+class ReservationRepositoryTest {
```
RepositoryTest 좋습니다 👍  다만 제일 복잡해보이는 `findReservationsByUserAndThemeAndFromAndTo` 메서드는 없네요 🤔 전체적으로 부족한 테스트 채워주면 더 좋을 것 같습니다!

KJungW replied:
전반적으로 부족한 테스트를 모두 채워넣어 보겠습니다!

summerlunaa reviewed (changes requested):
랜디 안녕하세요! 파랑입니다 🐳 
레벨2에서 다시 만났네요!! 너무 반가워요~~ 작성하신 코드와 남겨주신 질문에 대해 코멘트 드렸습니다.
혹시 어떤 주제든 질문이 있거나 리뷰에서 이해 안 되는 부분 있으면 언제든 편하게 DM 주셔도 괜찮습니다~ 이번 미션도 파이팅 해보시죠 💪

Commit:
test: 불필요한 TestDataConfig들 제거

Cursor AI를 활용해 대략적으로 TestDataConfig를 제거하고
대신 TestEntityManager를 통해 테스트 초기데이터 세팅

Commit:
refactor: 패키지 구조 변경

Commit:
refactor: 테스트 패키지 구조 수정

Commit:
refactor: 도메인 정리

Commit:
refactor: 로직 정리

Commit:
refactor: DTO 분리

Commit:
refactor: JPQL 쿼리 오류 수정

Commit:
test: 레파지토리 테스트 추가

Commit:
refactor: 인증 관련 로직 정리

Commit:
refactor: 예약 관련 로직 정리

Commit:
refactor: 예약 시간 조회 기능 단순화

Commit:
refactor: 테마 삭제 검증 추가

Commit:
refactor: 유저 추가 기능 리턴값 변경

Commit:
test: 서비스 계층 테스트 추가

Commit:
test: JwtTokenProvider 단위 테스트 추가

Commit:
refactor: 토큰 매개변수 리졸버 로직 변경

애노테이션을 통해 쿠키에 담긴 토큰 정보를 API 메서드로
바로 받을 수 있도록 로직 수정

Commit:
fix: 토큰 매개변수 리졸빙 로직 버그 수정

Commit:
refactor: 어드민 요청 인터셉터 로직 단순화

Commit:
refactor: CookieUtility 로직 변경

Commit:
test: CookieUtility 테스트 추가

Commit:
test: dto 테스트 추가

Commit:
test: 엔티티 관련 테스트 추가

Commit:
refactor: 엔티티 필드에 각종 제약 조건 추가

Commit:
refactor: id가 null인 엔티티의 비교 연산

Commit:
refactor: 유저 추가시 이메일 검증 추가

Commit:
refactor: 회원과 권한명 수정

Commit:
refactor: 클라이언트 코드 동기화

KJungW commented:
프로젝트에 대해 많은 리펙토링을 진행했습니다!

- ~~TestDataConfig로 초기 데이터 세팅하는 부분 제거 (반복적인 부분으로 Cursor AI를 활용)
	- 테스트 코드의 목적은 빠르게 문제의 원인을 파악하고 고치는 것이라고 생각
	- 하지만 ~~TestDataConfig으로 인해 테스트 코드 파악이 복잡하다고 판단
        - 최종적으로 테스트 메서드에서 TestEntityManager를 통해 자신의 테스트 환경을 세팅하도록 수정
- 패키지 구조 단순화
	- 도메인 별로 컨트롤러와 서비스를 묶어 패키지를 구성하는 것도 충분한 이점이 있다고 판단
	- 하지만 현재 프로젝트 크기에서는 도메인 별로 패키지를 구성하는 것이 오히려 설계 구조를 파악하기 어렵다고 느낌
	- 결과적으로는 패키지 구조를 단순화하는 것으로 결정
- 서비스가 서비스를 참조하지 않도록 설정
	- 순환참조 문제를 방지하기 위해 서비스가 서비스를 참조하지 않도록 설정
	- 이후 프로젝트가 커지고 필요성이 생기면 상위 서비스와 하위 서비스로 나누어 상위 서비스가 하위 서비스를 바라보도록 설정
- DTO를 분리
	- Request DTO는 오직 API 요청 용도로만 사용하고, Response DTO는 오직 API 응답 용도로만 사용
	- controller-service-repository 사이에 데이터를 운반하는 DTO는 별도로 관리
	- dto값 변경으로 예상치 못하게 API 요청/응답이 변경되는 것을 방지하기 위해 위와 같은 방식으로 DTO 분리
- 일부 Fixture 제거
	- 단순히 생성자나 생성 메서드를 호출해 데이터를 생성하는 Fixture 제거
	- 테스트를 확인하기 위해 불필요한 Depth를 하나 늘리는 것이라고 판단했음
- 각종 로직 정리 및 부족한 검증 로직 추가
- 부족한 테스트를 전체적으로 추가
- 전체적인 코드 정리

아직 리펙토링 해볼만한 점이 많지만 미션 일정을 생각해 먼저 리뷰를 받고 진행해보려고 합니다!

- request dto에 대한 빈검증 추가
- 서버 애플리케이션에 대한 E2E 테스트 추가
- 커스텀 예외 단순화
- ProblemDetail을 활용한 예외응답 형식 추가

summerlunaa commented on src/main/java/roomescape/domain/Member.java:
```diff
@@ -0,0 +1,88 @@
+    @Enumerated(value = EnumType.STRING)
+    @Column(nullable = false)
+    private Role role;
+    @Column(nullable = false, length = 255)
```
length는 기본값이 255라 특별히 다른 경우 아니라면 굳이 명시하지 않아도 될 것 같긴하네요ㅎㅎㅎ
사실 문자열의 길이 제약이 100자여도 DB 제약은 넉넉하게 해두는 편이거든요. DB는 변경이 매우 어려운데 서비스의 요구사항은 변경이 잦으니까요. 즉, DB의 문자열 길이는 크게 의미 없고, 문자열 길이 제약이 필요하다면 어플리케이션에서 검증하는 것이 좋을 것 같습니다!

KJungW replied:
당연히 문자열 길이 제약이 애플리케이션과 DB가 모두 동일해야한다고 관성적으로 생각하고 있었는데 새로운 관점을 알려주셔서 감사합니다! 이번 프로젝트에도 한번 적용해보겠습니다!

summerlunaa commented on src/main/java/roomescape/dto/business/ReservationTimeWithBookCount.java:
```diff
@@ -0,0 +1,11 @@
+
+import java.time.LocalTime;
+
+public record ReservationTimeWithBookCount(
```
사용처가 없네요 👀

KJungW replied:
앗.. 바로 삭제하겠습니다!

summerlunaa commented on src/test/java/roomescape/controller/AdminControllerTest.java:
```diff
@@ -0,0 +1,105 @@
+//package roomescape.controller;
+//
+//import static roomescape.test.fixture.DateFixture.NEXT_DAY;
+//
+//import io.restassured.RestAssured;
```
주석 처리된 테스트들 확인 부탁드려요 😢

KJungW replied:
컨트롤러에 대한 테스트는 다음 단계에서 작성할 E2E 통합테스트로 대체해서 작성하겠습니다. 현재 컨트롤러 테스트는 제거하겠습니다!

summerlunaa reviewed (changes requested):
랜디 안녕하세요!
리뷰 반영 잘 해주셨습니다 👍  마지막으로 코멘트 조금 더 남겼어요.
확인해 보시고 빠르게 리뷰 요청 주세요~ 오래 걸리는 건 다음 단계로 넘기시죠ㅎㅎ 다음 번엔 바로 머지할게요!

Commit:
refactor: 회원의 예약 조회 기능 이동

Commit:
refactor: 서비스 메서드 컨벤션 변경

Commit:
refactor: user 키워드 수정

Commit:
refactor: 클라이언트 코드 동기화

Commit:
refactor: 각종 환경 설정값 추가

Commit:
refactor: 도메인 검증 추가

Commit:
refactor: 불필요한 클래스 제거

summerlunaa reviewed (approved):
랜디 안녕하세요!
이번에도 리뷰 반영 잘 해주셨습니다 👍 가능하면 테스트를 주석처리하는 일은 없었으면 좋겠지만 ㅎㅎ 이제 다음 단계로 넘어가야 하니 E2E 통합테스트는 3-4단계에서 확인할게요.
3-4단계도 파이팅입니다 💪
